internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -190,6 +198,16 @@ func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkf
 
 	data, err := ws.getSessionData(wctx, params.SessionID)
 	if err == nil {
+		if workflow.IsReplaying(wctx) && data.Session.State().IsFinal() {
+			// HACK: If we somehow get a signal to this workflow after it completed,
+			//       in certain delicate timing, the workflow rekicks in replay.
+			//       Though replay would work just fine, and result in a no-op,
+			//       it's nice to not have to run through its entirely. Temporal
+			//       just seems to ignore it...
+			z.Warn("already completed")
```

Comment:
It's not a warn, the system would keep working as expected with this fix
Info at best

Reply:
yeah... it's still odd and i'd like to get rid of it. but you're right, I'll change to info.